app/services/find_or_register_user.rb

```diff
@@ -0,0 +1,27 @@
+class FindOrRegisterUser
```

Comment:
`RegisterUser` works for me -- that encapsulates the idea of finding or creating. Probably a smell to have the word "or" in a class name anyway since that indicates breakage of Single Responsibility Principle